rust-runtime/aws-smithy-http-server/src/extension.rs

```diff
@@ -34,10 +34,51 @@
 
 //! Extension extraction to share state across handlers.
 
-use super::rejection::{ExtensionRejection, ExtensionsAlreadyExtracted, MissingExtension};
+use super::rejection::{ExtensionHandlingRejection, ExtensionsAlreadyExtracted, MissingExtension};
 use async_trait::async_trait;
 use axum_core::extract::{FromRequest, RequestParts};
-use std::ops::Deref;
+use std::{borrow::Cow, ops::Deref};
+
+/// Extension type used to store the Smithy model namespace.
+#[derive(Debug, Clone)]
+pub struct ExtensionNamespace(pub Cow<'static, str>);
+
+/// Extension type used to store the Smithy operation name.
+#[derive(Debug, Clone)]
+pub struct ExtensionOperationName(pub Cow<'static, str>);
+
+/// Extension type used to store the type of framework error caught during execution.
+/// These are unmodeled error, or rejection, defined in the runtime crates.
+#[derive(Debug, Clone)]
+pub struct ExtensionRejection(pub Cow<'static, str>);
+
+/// Extension type used to store the type of user defined error returned by an operation.
+/// These are modeled errors, defined in the Smithy model.
+#[derive(Debug, Clone)]
+pub struct ExtensionModeledError(pub Cow<'static, str>);
+
+/// Implement new for all Extension holding a Cow<'static, str>.
+macro_rules! impl_extension_new_static {
+    ($name:ident, $sname:expr) => {
+        impl $name {
+            #[doc = "Returns a new `"]
+            #[doc = $sname]
+            #[doc = "`."]
+            pub fn new<S: Into<String>>(value: S) -> $name {
+                $name(std::borrow::Cow::from(value.into()))
+            }
+        }
+    };
+
+    ($name:tt) => {
```

Comment:
Better to use `concat!` for this: https://stackoverflow.com/a/68783193

Reply:
Didn't know, will check.